rust-runtime/aws-smithy-http-server/src/routing/routers/aws_json.rs

```diff
@@ -0,0 +1,128 @@
+use std::convert::Infallible;
+
+use http::header::ToStrError;
+use tower::{Layer, Service};
+
+use crate::{
+    body::BoxBody,
+    extension::RuntimeErrorExtension,
+    protocols::{AwsJson10, AwsJson11},
+    response::IntoResponse,
+    routing::{tiny_map::TinyMap, Route},
+};
+
+use super::Router;
+
+pub enum Error {
+    NotRootUrl,
+    MethodDisallowed,
+    MissingHeader,
+    InvalidHeader(ToStrError),
+    NotFound,
+}
+
+impl IntoResponse<AwsJson10> for Error {
+    fn into_response(self) -> http::Response<BoxBody> {
+        match self {
+            Error::MethodDisallowed => super::method_disallowed(),
+            _ => http::Response::builder()
+                .status(http::StatusCode::NOT_FOUND)
+                .header("Content-Type", "application/x-amz-json-1.0")
+                .extension(RuntimeErrorExtension::new(
+                    super::UNKNOWN_OPERATION_EXCEPTION.to_string(),
```

Comment:
You don't have to change it, but my preference for some time has been to use `to_owned()`, because of https://users.rust-lang.org/t/to-string-vs-to-owned-for-string-literals/1441/6

Curious to learn what you prefer and why.

Reply:
I prefer `to_string` over `into_owned` generally because the reader immediately knows that the output type is `String` when they read `s.to_string()` whereas with `s.to_owned()` you can't know without prior knowledge of `s`'s type. I guess if this was an inline string literal that wouldn't be a problem.

Reply:
Yeah, fair point, it's only applicable for string literals defined in place.